src/main/java/io/jenkins/pluginhealth/scoring/service/PluginService.java

```diff
@@ -44,6 +46,12 @@ public PluginService(PluginRepository pluginRepository) {
     public Plugin saveOrUpdate(Plugin plugin) {
         return pluginRepository.findByName(plugin.getName())
             .map(pluginFromDatabase -> pluginFromDatabase.setScm(plugin.getScm()).setReleaseTimestamp(plugin.getReleaseTimestamp()))
+            .map(pluginFromDatabase -> {
+                for (Map.Entry<String, ProbeResult> freshlyReadProbe : plugin.getDetails().entrySet()) {
+                    pluginFromDatabase.addDetails(freshlyReadProbe.getValue());
+                }
+                return pluginFromDatabase;
```

Comment:
Could we have a new method this would take a map and add its content to the object?